serde/src/private/de.rs

```diff
@@ -1496,14 +1496,6 @@ mod content {
             drop(self);
             visitor.visit_unit()
         }
-
-        fn __deserialize_content<V>(self, visitor: V) -> Result<V::Value, Self::Error>
-        where
-            V: Visitor<'de, Value = Content<'de>>,
-        {
-            let _ = visitor;
-            Ok(self.content)
-        }
```

Comment:
This is not needed anymore because you can use it in another crate and keep this optimization

Reply:
I'm a bit unclear on that tbh; won't we have to keep `Content` in `serde_core` to make this work?

Reply:
@dtolnay brought up https://github.com/serde-rs/serde/pull/2912 when I last spoke to him, but I guess this still requires `Content` to live in `serde_core`.
I may be clinging onto `Content` removal for no good reason, though to me it seems like a private implementation detail of derive machinery, which I guess we don't want to live in core? (per: https://github.com/serde-rs/serde/pull/2608#pullrequestreview-1899765113)